src/main/scala/io/iohk/ethereum/faucet/FaucetApi.scala

```diff
@@ -0,0 +1,83 @@
+package io.iohk.ethereum.faucet
+
+import java.time.Clock
+
+import akka.http.scaladsl.model.{RemoteAddress, StatusCodes}
+import akka.http.scaladsl.server.Route
+import akka.http.scaladsl.server.Directives._
+import akka.util.ByteString
+import ch.megard.akka.http.cors.scaladsl.CorsDirectives.cors
+import ch.megard.akka.http.cors.scaladsl.settings.CorsSettings
+import com.twitter.util.LruMap
+import de.heikoseeberger.akkahttpjson4s.Json4sSupport
+import io.iohk.ethereum.domain.{Address, Transaction}
+import io.iohk.ethereum.keystore.KeyStore
+import io.iohk.ethereum.mallet.service.RpcClient
+import io.iohk.ethereum.utils.Logger
+import io.iohk.ethereum.rlp
+import io.iohk.ethereum.network.p2p.messages.CommonMessages.SignedTransactions.SignedTransactionEnc
+import org.spongycastle.util.encoders.Hex
+
+class FaucetApi(
+    rpcClient: RpcClient,
+    keyStore: KeyStore,
+    config: FaucetConfig,
+    clock: Clock = Clock.systemUTC())
+  extends Json4sSupport
+    with Logger {
+
+  private val latestRequestTimestamps = new LruMap[RemoteAddress, Long](config.latestTimestampCacheSize)
+
+  private val wallet = keyStore.unlockAccount(config.walletAddress, config.walletPassword) match {
+    case Right(w) => w
+    case Left(err) => throw new RuntimeException(s"Cannot unlock wallet for use in faucet (${config.walletAddress}), because of $err")
+  }
+
+  private val corsSettings = CorsSettings.defaultSettings.copy(
+    allowGenericHttpRequests = true,
+    allowedOrigins = config.corsAllowedOrigins)
+
+  val route: Route = cors(corsSettings) {
+    (path("faucet") & pathEndOrSingleSlash & post & parameter('address)) { targetAddress =>
+      extractClientIP { clientAddr =>
+        handleRequest(clientAddr, targetAddress)
+      }
+    }
+  }
+
+  private def handleRequest(clientAddr: RemoteAddress, targetAddress: String) = {
+    val timeMillis = clock.instant().toEpochMilli
+    val latestRequestTimestamp = latestRequestTimestamps.getOrElse(clientAddr, 0L)
+    if (latestRequestTimestamp + config.minRequestInterval.toMillis < timeMillis) {
+      latestRequestTimestamps.put(clientAddr, timeMillis)
+
+      val res = for {
+        nonce <- rpcClient.getNonce(wallet.address)
+        txId <- rpcClient.sendTransaction(prepareTx(Address(targetAddress), nonce))
+      } yield txId
+
+      res match {
+        case Right(txId) =>
+          complete(StatusCodes.OK, s"0x${Hex.toHexString(txId.toArray[Byte])}")
```

Comment:
Let's  add logging for the successful case, too. It will be interesting to see how often this is used (include IP and ETH addresses and transfer value)